util/netcdf_generator.py

```diff
@@ -83,6 +83,37 @@ def _filter_params(self, ds, params):
                 ds = ds.drop(key)
         return ds
 
+    def _setup_coordinate_variables(self, ds):
+        """
+        Find the depth variable in the dataset and set assocciated coordinate attributes (10745)
+        :param ds: the dataset on which to operate
+        :return: the dataset with the coordinate variables and associated attributes setup
+        """
+        # find the depth_variable (10745 AC1)
+        depth_variable = find_depth_variable(ds.data_vars)
+        coordinates_key = 'coordinates'
+        coordinate_variables = "time lat lon"
+        if depth_variable:
+            coordinate_variables += " " + depth_variable
+        for var in ds.data_vars:
+            # coordinate variable shouldn't have coordinate attribute (10745 AC3)
+            # only scientific variables should have coordinate attribute (10745 AC2)
+            if var not in coordinate_variables \
+                and var not in app.config["NETCDF_NONSCI_VARIABLES"]:
+                ds[var].attrs[coordinates_key] = coordinate_variables
+            elif coordinates_key in ds[var].attrs:
+                del ds[var].attrs[coordinates_key]
+            # make sure coordinate variables have axis defined (10745 AC4)
+            if var == 'lat':
+                ds[var].attrs['axis'] = 'Y'
+            elif var == 'lon':
+                ds[var].attrs['axis'] = 'X'
+            elif var == depth_variable:
+                ds[var].attrs['axis'] = 'Z'
+
```

Comment:
Does 'time' already consistently have axis defined as 'T', or should it be added to this section?

Reply:
time is consistently handled by code in datamodel.py's to_xray_dataset function.  analysis indicates this function gets called on all products (it's run against any data coming back from cassandra) and doesn't need to be added here.

Reply:
as a side note, I initially had this code in the to_xray_dataset function, but found out that at that point the product doesn't include any derived or external parameters, so I had to move it further down the chain into netcdf_generator when the product was in a more final state.